src/coprocessor/endpoint.rs

```diff
@@ -418,10 +418,24 @@ impl<E: Engine> Endpoint<E> {
         let snapshot =
             unsafe { with_tls_engine(|engine| Self::async_snapshot(engine, &tracker.req_ctx)) }
                 .await?;
+        let latest_buckets = snapshot.ext().get_buckets();
+
+        // Check if the buckets version is latest.
+        // skip if request don't carry this bucket version.
+        if let Some(ref buckets) = latest_buckets&&
+            buckets.version > tracker.req_ctx.context.buckets_version &&
+            tracker.req_ctx.context.buckets_version!=0 {
+                let mut bucket_not_match = errorpb::BucketVersionNotMatch::default();
+                bucket_not_match.set_version(buckets.version);
+                bucket_not_match.set_keys(buckets.keys.clone().into());
```

Comment:
Is there any size limit of coprocessor response? Can it cause error `grpc: received message larger than max ...`?

Reply:
the grpc send limit: Maximum value: 2147483647. buckets keys can't exceed than it even than 100GB/50MB=2000 kets .If happened, the buckets report also met this problem.